notifications/emailer/emailer.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package emailer
+
+import (
+	"fmt"
+
+	"github.com/absmach/supermq/internal/email"
+	"github.com/absmach/supermq/notifications"
+)
+
+var _ notifications.Emailer = (*emailer)(nil)
```

Comment:
you should have a notifier interface and implmentations  for different notifiers (SMS, email, slack ...)